misk/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -0,0 +1,127 @@
+package misk.scope
+
+import com.google.inject.Key
+import com.google.inject.Provider
+import java.util.concurrent.Callable
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KCallable
+import kotlin.reflect.KFunction
+import kotlin.reflect.KParameter
+import kotlin.reflect.KType
+import kotlin.reflect.KTypeParameter
+import kotlin.reflect.KVisibility
+
+@Singleton
+class ActionScope @Inject internal constructor(
+        // NB(mmihic): ActionScoped depends on ActionScope depends on
```

Comment:
somewhat off topic: with TODOs we put a username in to volunteer who will do it. For `NB` notes is there any reason to have a username?

Reply:
It's the name of the person that has context for the NB. Could get this from git blame but that might not survive refactorings.